### PR TITLE
1. Handling compression, 2. Using charset from headers

### PR DESCRIPTION
**Pull request by ssesha**
js resources are mostly served compressed from cdns eg. http://sd1.gemvara.net/static/bundle-bundle_site_head_39e0ef67d1c.js

### CODE REVIEW

**Comment by domenic** [comment]
This isn't quite the right approach to this. We should just use the request package, instead of http/https.request. It already takes care of this stuff and we're already using it in a few places. It would then also handle redirects etc.

**Comment by inikulin** [comment]
**FIXME**: request doesn't deal with the charset encoding. Seems like we need to use something like https://github.com/ashtuchkin/iconv-lite to deal with it + implement [this thing](http://www.w3.org/TR/html5/syntax.html#determining-the-character-encoding) in the parse5.

**Comment by ssesha** [comment]
@domenic I have made the change to use the request module instead of http/https. I see that we get redirects and gzipped response handling taken care of. Haven't handled different charset encodings though. Can you review this?

File: lib/jsdom/level2/html.js [hunk 1]
```diff
@@ -8,8 +8,8 @@ var core                  = require("../level1/core"),
     URL                   = require("url"),
     Path                  = require('path'),
     fs                    = require("fs"),
-    http                  = require('http'),
-    https                 = require('https');
+    zlib                  = require('zlib'),
```
Comment (domenic) [inline lib/jsdom/level2/html.js]: No longer needed

**Comment by domenic** [comment]
LGTM with one nit. Will merge (and leave a more detailed comment about encoding) when I get back to my computer for a few hours. Thanks!

**Comment by ssesha** [comment]
removed the unused module

**Comment by domenic** [comment]
Merged! Now to decide whether this is a bugfix or a feature...

**Comment by ssesha** [comment]
Lets call it a feature, feature sounds cool

**Comment by domenic** [comment]
Yeah, I went with a minor release https://github.com/tmpvar/jsdom/blob/master/Changelog.md#440 :)

**Pull request merged**